utils/hct/hctdb.py

```diff
@@ -5508,10 +5511,38 @@ def UFI(name, **mappings):
         )
         next_op_idx += 1
 
+        # RayQuery
+        self.add_dxil_op(
+            "AllocateRayQuery2",
+            next_op_idx,
+            "AllocateRayQuery2",
+            "allocates space for RayQuery and return handle",
+            "v",
+            "",
+            [
+                db_dxil_param(0, "i32", "", "handle to RayQuery state"),
+                db_dxil_param(
+                    2,
+                    "u32",
+                    "constRayFlags",
+                    "Valid combination of RAY_FLAGS",
+                    is_const=True,
+                ),
+                db_dxil_param(
+                    3,
+                    "u32",
+                    "constRayQueryFlags",
+                    "Valid combination of RAYQUERY_FLAGS",
+                    is_const=True,
+                ),
+            ],
+        )
+        next_op_idx += 1
+
         # End of DXIL 1.8 opcodes.
```

Comment:
This intrinsic should have been added after this marker.  This is a DXIL 1.9 opcode.

Reply:
fyi, https://github.com/microsoft/DirectXShaderCompiler/pull/7125 has been updated to reserve the AllocateRayQuery2 opcode in DXIL 1.9